receiver/nsxreceiver/README.md

```diff
@@ -0,0 +1,97 @@
+# NSX-T Receiver
+
+This receiver fetches metrics important to run virtual networking using NSX-T. The receiver ingests metrics via the [NSX Rest API](https://docs.vmware.com/en/VMware-NSX-Data-Center-for-vSphere/6.4/nsx_64_api.pdf). As for logs, it is encouraged to use the [syslogreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/syslogreceiver) which can be used to ingest NSX forwarded logs. Please see [Logging-Configuration](#logging-configuration) for more information.
+
+Supported pipeline types: `metrics`
+
+> :construction: This receiver is in **BETA**. Configuration fields and metric data model are subject to change.
+
+## Purpose
+
+The purpose of this receiver is to allow users to monitor metrics from NSX-T environments.
+
+## Prerequisites
+
+Needs at least an “Auditor” user for NSX Manager Resources in the vSphere Client (can be created via Web UI or CLI)
+
+The collector must be able to reach the NSX Manager with port 443 open.
+
+This receiver supports NSX-T Datacenter versions:
+
+- 3.2.0
+- 3.1.2
+
+## Metrics Configuration
+
+Under the `metrics` key of the NSX configuration are these parameters:
+
+- `settings` (default: see DefaultMetricsSettings [here])(./internal/metadata/generated_metrics_v2.go): Allows enabling and disabling specific metrics from being collected in this receiver.
+
+- `endpoint`: Endpoint of the NSX Manager. Must be formatted as `{scheme}://{host}:{port}`. Schems supported are `http` and `https`
+
+- `username`: Username of the `Network Admin` user
+
+- `password`: Password of the `Network Admin` user
+
+- `collection_interval`: (default = `1m`): This receiver collects metrics on an interval. This value must be a string readable by Golang's [time.ParseDuration](https://pkg.go.dev/time#ParseDuration). Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
+
+- `timeout`: (default = `1m`) The timeout of running commands against the NSX REST API.
+
+### Example Configuration
+
+```yaml
+receivers:
+  nsx:
+    endpoint: https://nsx-manager
+    username: admin
+    password: password
+    timeout: 60s
+    settings:
```

Comment:
Since this receiver will be metric only, this should be called `metrics` in order to match other receivers.